lint/cadence_v1_analyzer.go

```diff
@@ -0,0 +1,330 @@
+				Report()
+		case "PublicAccount":
+			v.
+				newDiagnostic(
+					typeAnnotation.Type,
+					"`PublicAccount` has been removed in Cadence 1.0.  Please use an `&Account` reference instead.",
+					"C1.0-PublicAccount",
+					"https://forum.flow.com/t/update-on-cadence-1-0/5197#account-access-got-improved-55",
+				).
+				WithSimpleReplacement("&Account").
+				Report()
+		}
+	})
+}
+
+func (v *cadenceV1Analyzer) analyzeResourceDestructors() {
+	v.inspector.WithStack(
+		[]ast.Element{
+			(*ast.SpecialFunctionDeclaration)(nil),
+		},
+		func(element ast.Element, push bool, stack []ast.Element) bool {
+			declaration := element.(*ast.SpecialFunctionDeclaration)
+			if declaration.DeclarationIdentifier().Identifier != "destroy" {
+				return true
+			}
+
+			parent, ok := stack[len(stack)-2].(*ast.CompositeDeclaration)
+			if !ok {
+				return true
+			}
+
+			if parent.Kind() != common.CompositeKindResource {
+				return true
+			}
+
+			v.reportRemovedResourceDestructor(declaration)
+			return false
+		},
+	)
+}
+
+func (v *cadenceV1Analyzer) reportRemovedResourceDestructor(
+	declaration *ast.SpecialFunctionDeclaration,
+) {
+	shouldSuggestRemoval := func() bool {
+		functionDeclaration := declaration.FunctionDeclaration
+		if !functionDeclaration.FunctionBlock.HasStatements() {
+			return true
+		}
+
+		for _, statement := range functionDeclaration.FunctionBlock.Block.Statements {
+			expressionStatement, ok := statement.(*ast.ExpressionStatement)
+			if !ok {
+				return false
+			}
+
+			if _, ok := expressionStatement.Expression.(*ast.DestroyExpression); !ok {
+				return false
+			}
+		}
+
+		return true
+	}
+
+	diagnostic := v.newDiagnostic(
+		declaration,
+		"`destroy` keyword has been removed.  Sub-resources will now be implicitly destroyed with their parent.  A `ResourceDestroyed` event can be configured to be emitted to notify clients of the destruction.",
+		"C1.0-ResourceDestruction",
+		"https://forum.flow.com/t/update-on-cadence-1-0/5197#force-destruction-of-resources-101",
+	)
+
+	if shouldSuggestRemoval() {
+		diagnostic.WithSimpleReplacement("")
+	}
+	diagnostic.Report()
+}
+
+// Type annotations are not part of traversal, so we need to inspect them separately
+func (v *cadenceV1Analyzer) inspectTypeAnnotations(f func(typeAnnotation *ast.TypeAnnotation)) {
+	// Filter out nil type annotations
+	processAnnotation := func(t *ast.TypeAnnotation) {
+		if t != nil {
+			f(t)
+		}
+	}
+
+	// Helper function to process a parameter list
+	processParameterList := func(parameterList *ast.ParameterList) {
+		for _, parameter := range parameterList.Parameters {
+			processAnnotation(parameter.TypeAnnotation)
+		}
+	}
+
+	v.inspector.Preorder(
+		[]ast.Element{
+			(*ast.FieldDeclaration)(nil),
+			(*ast.FunctionExpression)(nil),
+			(*ast.CastingExpression)(nil),
+			(*ast.FunctionDeclaration)(nil),
+			(*ast.TransactionDeclaration)(nil),
+			(*ast.VariableDeclaration)(nil),
+			(*ast.SpecialFunctionDeclaration)(nil),
+		},
```

Comment:
Also check `InvocationExpression`'s type arguments and `InstantiationType`'s type arguments.